### PR TITLE
Update firebase to latest version

### PR DESCRIPTION
**Pull request by mueller-ma**
Signed-off-by: mueller-ma <mueller-ma@users.noreply.github.com>

### CODE REVIEW

File: mobile/src/full/java/org/openhab/habdroid/core/FcmRegistrationWorker.kt [hunk 1]
```diff
@@ -119,8 +118,7 @@ class FcmRegistrationWorker(private val context: Context, params: WorkerParamete
     // HttpException is thrown by our HTTP code, IOException can be thrown by FCM
     @Throws(HttpClient.HttpException::class, IOException::class)
     private suspend fun registerFcm(connection: CloudConnection) {
-        val token = FirebaseInstanceId.getInstance().getToken(connection.messagingSenderId,
-                FirebaseMessaging.INSTANCE_ID_SCOPE)
```
Comment (mueller-ma) [inline mobile/src/full/java/org/openhab/habdroid/core/FcmRegistrationWorker.kt:123]: I didn't find a way to pass `connection.messagingSenderId` to Firebase.

Comment (maniac103) [reply mobile/src/full/java/org/openhab/habdroid/core/FcmRegistrationWorker.kt:123]: If it works nevertheless, that's not a problem I guess. What *is* a problem, AFAICT, is that `token` will have type `Task<String>`, not `String`, that is, you'll need to add a listener to retrieve the actual token.

Comment (mueller-ma) [reply mobile/src/full/java/org/openhab/habdroid/core/FcmRegistrationWorker.kt:123]: I haven't tested this yet. I added `.result` which returns a `String`.

Comment (maniac103) [reply mobile/src/full/java/org/openhab/habdroid/core/FcmRegistrationWorker.kt:123]: > .result which returns a String.

... or throws an `IllegalStateException` if the task isn't done yet ;-)

Comment (mueller-ma) [reply mobile/src/full/java/org/openhab/habdroid/core/FcmRegistrationWorker.kt:123]: Fetching the token now works and the registration at myopenhab is successful, but notifications aren't received.

Comment (digitaldan) [reply mobile/src/full/java/org/openhab/habdroid/core/FcmRegistrationWorker.kt:123]: I just tested this with my upcoming changes to openHAB cloud, and i can confirm it works and is getting notifications.  Once i make the change to the server, both the older code and this code will work.  The only gotcha is that users will need to launch the app to register again regardless of this change as we are using a new FCM id from our openhab-mobile project and not the original one.

**Comment by mueller-ma** [comment]
I removed the broken 'hideNotification' feature for now. It has to be replaced by the new endpoint in myopenHAB. As that is free software, it can be used in the foss flavor as well: #3686

**Pull request merged**